File: PYTHON/Pertemuan 01/MobilListrik.py
import sys
sys.stdout.reconfigure(encoding='utf-8')
from MobilWarisan import MobilWarisan

class MobilListrik(MobilWarisan):
    def __init__(self, merek: str, baterai_awal: float):
        super().__init__(merek, 4)
        self.baterai_persen = baterai_awal

    def cas(self):
        self.baterai_persen = 100.0
        print(f"[MobilListrik:{self.merek}] Baterai penuh: 100%")

    def jalan(self):
        if self.baterai_persen > 0:
            self.kecepatan = 80
            self.baterai_persen -= 10
            print(f"[MobilListrik:{self.merek}] Melaju {self.kecepatan} km/j | Baterai: {self.baterai_persen}%")
        else:
            print(f"[MobilListrik:{self.merek}] Baterai habis! Perlu di-charge.")

File: JAVA/Pertemuan 01/src/id/ac/uph/pbo/pertemuan01/MobilListrik.java
package id.ac.uph.pbo.pertemuan01;

public class MobilListrik extends MobilWarisan {
    private double bateraiPersen;

    public MobilListrik(String merek, double bateraiAwal) {
        super(merek, 4);
        this.bateraiPersen = bateraiAwal;
    }

    public void cas() {
        bateraiPersen = 100.0;
        System.out.println("[MobilListrik:" + merek + "] Baterai penuh: 100%");
    }

    @Override
    public void jalan() {   // Override method dari Kendaraan
        if (bateraiPersen > 0) {
            kecepatan      = 80;
            bateraiPersen -= 10;
            System.out.println("[MobilListrik:" + merek + "] Melaju " + kecepatan
                    + " km/j | Baterai: " + bateraiPersen + "%");
        } else {
            System.out.println("[MobilListrik:" + merek + "] Baterai habis! Perlu di-charge.");
        }
    }
}
